src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,132 @@
+     * @param isDone The task's status.
+     * @param byDate The task's due date and time.
+     */
+    public Deadline(String description, boolean isDone, LocalDate byDate) {
+        super(description, isDone);
+        this.byDate = byDate;
+        this.byDateTime = null;
+    }
+
+    /**
+     * Creates a `Deadline` task with description, status, due date and time.
+     *
+     * @param description The task's description.
+     * @param isDone The task's status.
+     * @param byDateTime The task's due date and time.
+     */
+    public Deadline(String description, boolean isDone, LocalDateTime byDateTime) {
+        super(description, isDone);
+        this.byDate = null;
+        this.byDateTime = byDateTime;
+    }
+
+    /**
+     * Creates a string representing a number with at least two digits.
+     *
+     * @param s The input number
+     * @return String representing the input number with at least two digits.
+     */
+    private String padZero(int s) {
+        if (s < 10) {
+            return "0" + s;
+        }
+        return Integer.toString(s);
+    }
+
+    /**
+     * Creates a formatted string to store the due date and time.
+     *
+     * @return The formatted string to store the due date and time.
+     */
+    public String getDeadlineForSave() {
+        return this.byDate == null
+                ? (this.byDateTime.getYear() + "-" + padZero(this.byDateTime.getMonthValue())
+                + "-" + padZero(this.byDateTime.getDayOfMonth()) + " " + padZero(this.byDateTime.getHour()) + ":"
+                + padZero(this.byDateTime.getMinute()))
+                : (this.byDate.getYear() + "-" + padZero(this.byDate.getMonthValue())
+                + "-" + padZero(this.byDate.getDayOfMonth()));
+    }
+
+    /**
+     * Creates a formatted string to display the due date and time.
+     *
+     * @return The formatted string to display the due date and time.
+     */
+    public String getDeadlineForDisplay() {
+        return this.byDate == null
+                ? (this.byDateTime.getMonth() + " " + padZero(this.byDateTime.getDayOfMonth())
+                + " " + this.byDateTime.getYear() + " " + padZero((this.byDateTime.getHour() > 12
+                ? this.byDateTime.getHour() - 12 : this.byDateTime.getHour())) + ":"
+                + padZero(this.byDateTime.getMinute()) + (this.byDateTime.getHour() >= 12 ? "PM" : "AM"))
+                : (this.byDate.getMonth() + " " + padZero(this.byDate.getDayOfMonth())
+                + " " + this.byDate.getYear());
+    }
+
+    /**
+     * Returns the formatted string containing done status and description of the task.
+     *
+     * @return The formatted string containing done status and description of the task.
+     */
+    @Override
+    public String getDescriptionStatus() {
+        return "[D]" + (isDone ? "[X] " : "[ ] ") + this.description + " (by: "
+                + this.getDeadlineForDisplay() + ")";
+    }
```

Comment:
If you're overriding, you could actually create a toString() method in Task and then use that toString() method and modify it inside Deadline.

Reply:
I choose to create a new method named `getDescriptionStatus` instead of using the original `toString` inherit from Object class because I'm not sure how many customised strings I will need to make for the class along the way in future increments.